packages/docusaurus-plugin-content-blog/src/routes.ts

```diff
@@ -249,10 +264,70 @@ export async function buildAllRoutes({
     return [tagsListRoute, ...tagsPaginatedRoutes];
   }
 
+  function createAuthorsRoutes(): RouteConfig[] {
```

Comment:
The algo looks wrong, it should iterate on `authorsMap`

I'd use something like this:

```ts
async function createAuthorsRoutes({authorsMap, blogPosts, ...}) {

  const blogPostsByAuthorKey: Record<string,BlogPost[]> = groupBlogPostsByAuthorKey(); // TODO

  const authors = Object.values(authorsMap);

  return Promise.all([
    createAuthorListRoute();
    ...authors.flatMap(createAuthorPaginatedRoute);
  ]);

  // Maybe authors with page: false could even appear on the list? to be defined 🤷
  async function createAuthorListRoute() {
		return {
      path: "/authors",
      props: {
        authors: authors.map(toAuthorProp) // TODO
      };
    }
  }
  

  async function createAuthorPaginatedRoute(author) {
    if (!author.page) {
      return [];
    }
    const authorBlogPosts = blogPostsByAuthorKey[author.key];
    const pages = paginateBlogPosts(authorBlogPosts);
    return pages.map(({metadata,items}) => {
      return {
        path: metadata.permalink,
        modules: {
          items: blogPostItemsModule(items),
          sidebar: sidebarModulePath,
        },
        props: {
          author: toAuthorProp(author),
          listMetadata: metadata,
        },
      }
    })
  }

}
```